libr/bin/format/xtac/bin_xtac.c

```diff
@@ -0,0 +1,587 @@
+			R_LOG_WARN ("Warning: read (xtac BLCK stub)\n");
+			free (blck_stub);
+			return false;
+		}
+		blck_stub->ptr_to_entry = p_blck_stub;
+		r_list_append (bin->blck_stubs, blck_stub);
+		if (p_blck_stub == blck_stub->ptr_to_next_entry) {
+			R_LOG_WARN ("Warning: an infinite loop is detected. Some header members of BOCK Stub might be broken\n");
+			break;
+		}
+		p_blck_stub = blck_stub->ptr_to_next_entry;
+		i++;
+		if (i >= max_depth) {
+			R_LOG_WARN ("Warning: Too many BLCK Stubs. Some header members of BLCK Stub might be broken\n");
+			break;
+		}
+	} while (p_blck_stub);
+	return true;
+}
+
+static bool r_bin_xtac_read_xtac_linked_list(RBinXtacObj *bin) {
+	if (!(bin->xtac_linked_list = r_list_newf(free))) {
+		r_sys_perror ("r_list_newf (r_bin_xtac_read_xtac_linked_list)");
+		return false;
+	}
+	RBinXtacLinkedListEntry *entry = NULL;
+	ut32 p_xtac_linked_list_entry = bin->header->ptr_to_xtac_linked_list_head;
+	do {
+		ut32 p_buffer = p_xtac_linked_list_entry;
+
+		if (!(entry = R_NEW0 (RBinXtacLinkedListEntry))) {
+			r_sys_perror ("R_NEW0 (xtac linked list)");
+			return false;
+		}
+		entry->ptr_to_entry = p_buffer;
+
+		if (r_buf_read_at (bin->b, p_buffer, (ut8*) entry, sizeof (ut32)) < 0) {
+			R_LOG_WARN ("Warning: read (xtac linked list metadata)\n");
+			free (entry);
+			return false;
+		}
+
+		const ut32 meta = GET_META_DATA(entry->meta_and_offset);
+		if (has_forward_edge_addr (meta)) {
+			p_buffer += sizeof (ut32);
+			if (r_buf_read_at (bin->b, p_buffer, (ut8*) &entry->forward_edge_addr, sizeof (ut32)) < 0) {
+				R_LOG_WARN ("Warning: read (xtac linked list forward edge address)\n");
+				free (entry);
+				return false;
+			}
+		}
+		if (has_backward_edge_addr (meta)) {
+			p_buffer += sizeof (ut32);
+			if (r_buf_read_at (bin->b, p_buffer, (ut8*) &entry->backward_edge_addr, sizeof (ut32)) < 0) {
+				R_LOG_WARN ("Warning: read (xtac linked list backward edge address)\n");
+				free (entry);
+				return false;
+			}
+		}
+
+		r_list_append(bin->xtac_linked_list, entry);
+		p_xtac_linked_list_entry += GET_OFFSET(entry->meta_and_offset) * 4;
+	} while (p_xtac_linked_list_entry < bin->header->ptr_to_addr_pairs);
+
+	if (GET_OFFSET(entry->meta_and_offset) != 0x0FFFFFFF) {
+		R_LOG_WARN ("Warning: xtac linked list is not properly terminated\n");
+		R_LOG_WARN ("Some entry of xtac linked list might be broken\n");
+	}
+
+	return true;
+}
+
+static bool r_bin_xtac_init(RBinXtacObj *bin) {
+	// NOTE: not tested on big-endian processor
+	if (!r_bin_xtac_read_header(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_address_pairs(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_module_name(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_nt_native_pathname(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_blck_stubs(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_xtac_linked_list(bin)) {
+		return false;
+	}
+
+	return true;
+}
+
+static void* r_bin_xtac_free(RBinXtacObj *bin) {
```

Comment:
```suggestion
static void r_bin_xtac_free(RBinXtacObj *bin) {
```
this function always return null, so just let it void